corehq/elastic.py

```diff
@@ -337,92 +337,8 @@ def fetch_all(initial_response):
 
 def es_query(params=None, facets=None, terms=None, q=None, es_index=None, start_at=None, size=None, dict_only=False,
              fields=None, facet_size=None):
-    if terms is None:
-        terms = []
-    if q is None:
-        q = {}
-    else:
-        q = copy.deepcopy(q)
-    if params is None:
-        params = {}
-
-    q["size"] = size if size is not None else q.get("size", SIZE_LIMIT)
-    q["from"] = start_at or 0
-
-    def get_or_init_anded_filter_from_query_dict(qdict):
-        and_filter = qdict.get("filter", {}).pop("and", [])
-        filter = qdict.pop("filter", None)
-        if filter:
-            and_filter.append(filter)
-        return {"and": and_filter}
-
-    filter = get_or_init_anded_filter_from_query_dict(q)
-
-    def convert(param):
-        #todo: find a better way to handle bools, something that won't break fields that may be 'T' or 'F' but not bool
-        if param == 'T' or param is True:
-            return 1
-        elif param == 'F' or param is False:
-            return 0
-        return param
-
-    for attr in params:
-        if attr not in terms:
-            attr_val = [convert(params[attr])] if not isinstance(params[attr], list) else [convert(p) for p in params[attr]]
-            filter["and"].append({"terms": {attr: attr_val}})
-
-    if facets:
-        q["facets"] = q.get("facets", {})
-        if isinstance(facets, list):
-            for facet in facets:
-                q["facets"][facet] = {"terms": {"field": facet, "size": facet_size or SIZE_LIMIT}}
-        elif isinstance(facets, dict):
-            q["facets"].update(facets)
-
-    if filter["and"]:
-        query = q.pop("query", {})
-        q["query"] = {
-            "filtered": {
-                "filter": filter,
-            }
-        }
-        q["query"]["filtered"]["query"] = query if query else {"match_all": {}}
-
-    if fields is not None:
-        q["fields"] = q.get("fields", [])
-        q["fields"].extend(fields)
-
-    if dict_only:
-        return q
-
-    es_index = es_index or 'domains'
-    es_interface = ElasticsearchInterface(get_es_new())
-    meta = ES_META[es_index]
-
-    try:
-        result = es_interface.search(meta.index, meta.type, body=q)
-        report_and_fail_on_shard_failures(result)
-    except ElasticsearchException as e:
-        raise ESError(e)
-
-    if fields is not None:
-        for res in result['hits']['hits']:
-            flatten_field_dict(res)
-
-    return result
-
-
-def stream_es_query(chunksize=100, **kwargs):
-    size = kwargs.pop("size", None)
-    kwargs.pop("start_at", None)
-    kwargs["size"] = chunksize
-    for i in range(0, size or SIZE_LIMIT, chunksize):
-        kwargs["start_at"] = i
-        res = es_query(**kwargs)
-        if not res["hits"]["hits"]:
-            return
-        for hit in res["hits"]["hits"]:
-            yield hit
+    # todo; remove after m4change code is deleted
```

Comment:
Apart from the one use that I replaced [above](https://github.com/dimagi/commcare-hq/pull/28380/files#diff-e8c29e4d83c4b54de5ea540784b53448R168), m4change custom reports use this, but SaaS is planning to delete m4change code and already has disconnected m4change app, so we can delete this right away. See https://github.com/dimagi/commcare-hq/pull/28098/ and https://dimagi-dev.atlassian.net/browse/SAAS-10957